consumatio/external/tmdb.py

```diff
@@ -9,47 +9,68 @@
+        query = f'https://api.themoviedb.org/3/tv/{tv_id}?api_key=c4bd02adee6e73c4d17e69b039267c90&language=en-US'
+        data = get_data(query)
         return tv_details_to_dict(data)
 
     def get_tv_providers(self, tv_id, country):
-        data = requests.get(f'https://api.themoviedb.org/3/tv/{tv_id}/watch/providers?api_key=c4bd02adee6e73c4d17e69b039267c90').json()
+        query = f'https://api.themoviedb.org/3/tv/{tv_id}/watch/providers?api_key=c4bd02adee6e73c4d17e69b039267c90' 
+        data = get_data(query)
         data = data['results'][country]
         return tv_providers_to_dict(data)
 
     def get_tv_images(self, tv_id):    
-        data = requests.get(f'https://api.themoviedb.org/3/tv/{tv_id}/images?api_key=c4bd02adee6e73c4d17e69b039267c90').json()
+        query = f'https://api.themoviedb.org/3/tv/{tv_id}/images?api_key=c4bd02adee6e73c4d17e69b039267c90'
+        data = get_data(query)
         return tv_images_to_dict(data)
 
     def get_season_details(self, tv_id, season_number):
-        data = requests.get(f'https://api.themoviedb.org/3/tv/{tv_id}/season/{season_number}?api_key=c4bd02adee6e73c4d17e69b039267c90').json()
+        query = f'https://api.themoviedb.org/3/tv/{tv_id}/season/{season_number}?api_key=c4bd02adee6e73c4d17e69b039267c90'
+        data = get_data(query)
         return season_details_to_dict(data, tv_id)
 
     def get_season_images(self, tv_id, season_number):
-        data = requests.get(f'https://api.themoviedb.org/3/tv/{tv_id}/season/{season_number}/images?api_key=c4bd02adee6e73c4d17e69b039267c90').json()
+        query = f'https://api.themoviedb.org/3/tv/{tv_id}/season/{season_number}/images?api_key=c4bd02adee6e73c4d17e69b039267c90' 
+        data = get_data(query)
         return season_images_to_dict(data)
 
     def get_episode_details(self, tv_id, season_number, episode_number):
-        data = requests.get(f'https://api.themoviedb.org/3/tv/{tv_id}/season/{season_number}/episode/{episode_number}?api_key=c4bd02adee6e73c4d17e69b039267c90').json()
+        query = f'https://api.themoviedb.org/3/tv/{tv_id}/season/{season_number}/episode/{episode_number}?api_key=c4bd02adee6e73c4d17e69b039267c90'
+        data = get_data(query)
         return episode_details_to_dict(data)
 
     def get_episode_images(self, tv_id, season_number, episode_number):
-        data = requests.get(f'https://api.themoviedb.org/3/tv/{tv_id}/season/{season_number}/episode/{episode_number}/images?api_key=c4bd02adee6e73c4d17e69b039267c90').json()
-        return episode_images_to_dict(data)
\ No newline at end of file
+        query = f'https://api.themoviedb.org/3/tv/{tv_id}/season/{season_number}/episode/{episode_number}/images?api_key=c4bd02adee6e73c4d17e69b039267c90'
+        data = get_data(query)
+        return episode_images_to_dict(data)
+
+def get_data(query):
+   db = Database()
+
+   if (db.is_cached(query)):
+      data = eval(db.get_from_cache(query))
```

Comment:
`eval()` can be quite dangerous. Instead please consider using `json.loads()`
```suggestion
      data = json.loads(db.get_from_cache(query))
```